### PR TITLE
Add pinning with 4EVERLAND

### PR DESCRIPTION
**Pull request by saullary**

### CODE REVIEW

**Review by bonustrack — COMMENTED** [review]
Thanks for your PR. Could you run the lint script? Also what should be used as EVER_BUCKET_NAME?

File: src/providers/4everland.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  region: "eu-west-2",
+  credentials: {
+    accessKeyId: config.apiKey,
+    secretAccessKey: config.apiSecret,
+  },
+});
+
+export async function set(data: Buffer | object) {
+  const start = Date.now();
+  const input = config;
```
Comment (bonustrack) [inline src/providers/4everland.ts:21]: Any reason to add config here? It doesn't seem to be used.

File: src/rpc.ts [hunk 2]
```diff
@@ -16,11 +17,13 @@ router.post('/', async (req, res) => {
     const size = Buffer.from(JSON.stringify(params)).length;
     if (size > MAX) return rpcError(res, 500, 'too large', id);
     const result = await Promise.any([
+      set4everland(params),
       setFleek(params),
       setInfura(params),
       setPinata(params),
       setWeb3Storage(params)
```
Comment (bonustrack) [inline src/rpc.ts:24]:
```suggestion
      setFleek(params),
      setInfura(params),
      setPinata(params),
      setWeb3Storage(params),
      set4everland(params)
```

**Comment by cryptobuffer** [comment]
> Thanks for your PR. Could you run the lint script? Also what should be used as EVER_BUCKET_NAME?

Since 4everland bucket is a aws-s3 compatible layer build on top of IPFS/AR, please create a bucket in https://dashboard.4everland.org/#/bucket/storage/ , and indicate it as EVER_BUCKET_NAME here

image

And you can create bucket auth token in `Settings - Auth Tokens - Bucket Auth Tokens`

image

File: src/upload.ts [hunk 3]
```diff
@@ -23,7 +24,7 @@ router.post('/upload', upload.single('file'), async (req, res) => {
       .webp({ lossless: true });
 
     const buffer = await fs.createReadStream(req.file.path).pipe(transformer).toBuffer();
-    const result = await Promise.any([setFleek(buffer), setInfura(buffer), setPinata(buffer)]);
+    const result = await Promise.any([set4everland(buffer), setFleek(buffer), setInfura(buffer), setPinata(buffer)]);
```
[the remark this answers is not in the document]
Comment (saullary) [reply src/upload.ts]: done

File: src/rpc.ts (hunk 2 above)
Comment (saullary) [reply src/rpc.ts]: done

**Pull request merged**